snap/info_snap_yaml.go

```diff
@@ -497,6 +505,24 @@ func setHooksFromSnapYaml(y snapYaml, snap *Info, strk *scopedTracker) {
 	}
 }
 
+func setSystemUsernamesFromSnapYaml(y snapYaml, snap *Info) error {
+	for user, data := range y.SystemUsernames {
```

Comment:
user can only have one value much later for now, but shouldn't we validate it already here anyway to conform to username expectations? basically use osutil.IsValidUsername already here

Reply:
This was done way earlier in validateInfoAndFlags(), called from checkSnap() in overlord/snapstate/check_snap.go which is defined as "checkSnap ensures that the snap can be installed" so by this time, everything should be good.